nanopub/profile.py

```diff
@@ -20,46 +14,86 @@ class ProfileError(RuntimeError):
     pass
 
 
-def get_orcid_id():
-    orcid_id = get_profile()[ORCID_ID]
-    if not orcid_id:
-        raise ProfileError('Your profile was not setup yet or not setup correctly. To setup '
-                           'your profile see instructions in Readme.md')
-    return orcid_id
+class Profile:
+    """Represents a user profile.
+
+    Attributes:
+        orcid_id: The user's ORCID
+        name: The user's name
+        public_key: Path to the user's public key
+        private_key: Path to the user's private key
+        nanopub_uri: URI of the user's profile nanopub
+    """
+    def __init__(
+            self,
+            orcid_id: str, name: str,
+            public_key: Path, private_key: Path,
+            nanopub_uri: Optional[str] = None
+            ) -> None:
+        """Create a Profile."""
+        self.orcid_id = orcid_id
+        self.name = name
+        self.public_key = public_key
+        self.private_key = private_key
+        self.nanopub_uri = nanopub_uri
+
+    @classmethod
+    def _yatiml_savorize(cls, node: yatiml.Node) -> None:
+        node.rename_attribute('profile_nanopub', 'nanopub_uri')
+
+    @classmethod
+    def _yatiml_sweeten(cls, node: yatiml.Node) -> None:
+        node.remove_attributes_with_default_values(cls)
```

Comment:
This removes `nanopub_uri` (or `profile_nanopub`) from the YAML file if it is None, which is what the original did as well.

Reply:
I think to simplify things we don't have to do that. Just leave it empty

Reply:
True, and it would save a line of code...

Reply:
I removed seasoning completely as discussed.